examples.md

```diff
@@ -0,0 +1,178 @@
+# Examples
+
+## Example 1: Guest Skeleton Usage
+
+The guest will implement the methods in [`wit/guest.wit`](/wit/guest.wit):
+- `configure()`, which will be called by the host to setup long-lived subscriptions, and receive any other implementor specific extensions/config..
+- `handler()`, which will be called by the host to handle messages sent to the channels subscribed to in `configure()`.
+
+As Wasm modules/components are mostly intended to be short-lived, the host is meant to create new instances for configuration and each message handling.
+
+### Rust
+
+```rust
+use crate::messaging_types::{GuestConfigurationResult, MessageResult};
+
+wit_bindgen::generate!({
+    path: "../wit",
+});
+
+struct MyGuest;
+
+impl guest::Guest for MyGuest {
+    fn configure() -> Result<GuestConfigurationResult, u32> {
+        // Configure the messaging system
+    }
+
+    fn handler(_ms: Vec<MessageResult>) -> Result<(), u32> {
+        // Handle the message
+    }
+}
+
+export_messaging!(MyGuest);
+```
+
+### C/C++
+
+```c
+#include "messaging.c"
+
+messaging_result_guest_configuration_error_t messaging_configure(void) {
+    // Configure the messaging system
+}
+
+messaging_result_void_error_t messaging_handle(messaging_list_message_t *message) {
+    // Handle the message
+}
+```
+
+### Go
+
+```go
+package main
+
+import (
+	msging "app/gen"
+)
+
+func init() {
+	a := MsgingImpl{}
+	msging.SetExportsWasiMessagingMessagingGuest(a)
+}
+
+type MsgingImpl struct {
+}
+
+func (e MessagingImpl) Configure msging.Result[msging.WasiMessagingConsumerGuestConfiguration, msging.WasiMessagingMessagingGuestError] {
+	// Configure the messaging system
+}
+
+func (e MessagingImpl) Handler msging.Result[struct{}, msging.WasiMessagingMessagingGuestError] {
+	// Handle the message
+}
+
+func main() {}
+```
+
+## Example 2: Guest Usage
+
+### Rust
+
+```rust
+use crate::messaging_types::{GuestConfigurationResult, MessageResult, connect, disconnect};
+
+wit_bindgen::generate!({
+    path: "../wit",
+});
+
+struct MyGuest;
+
+impl guest::Guest for MyGuest {
+    fn configure() -> Result<GuestConfigurationResult, u32> {
+        // This function will be called by the host, who will be maintaining a 
+        // long-lived client connection to a broker or other messaging system.
+        // The client will be subscribed to channels a, b, and c) with no extra configuration.
+        // As soon as configuration is set, the host should kill the Wasm instance.
+        Ok(GuestConfigurationResult {
+            channels: vec!["a".to_string(), "b".to_string(), "c".to_string()],
```

Comment:
This will depend on the implementation. 

~extra required information can be passed via `extensions`, and it will be the implementor's job to specify if they required any `extension` parameter to set these subscriptions.